src/mod/channels.mod/tclchan.c

```diff
@@ -1512,12 +1513,26 @@ static int tcl_channel_modify(Tcl_Interp *irp, struct chanset_t *chan,
       p = strchr(item[i], ':');
       if (p) {
         *p++ = 0;
-        *pthr = atoi(item[i]);
-        *ptime = atoi(p);
-        *--p = ':';
+        if ((!strtol(item[i], &endptr1, 10) && (strcmp(endptr1, "\0") != 0)) \
```

Comment:
The `\` at the end is unnecessary.
Could directly assign values to the `strtol` invocation and get rid of the `strcmp`: `if ((!(*pthr = (int) strtol(...)) && (*endptr)) || (!(*ptime = (int) strtol(...)) && (*endptr))) {`
The else-clause would be redundant then as long as you move the `*--p = ':';` out of it.